packages/web-components/src/components/ic-popover-menu/ic-popover-menu.tsx

```diff
@@ -0,0 +1,357 @@
+        break;
+      case "End":
+        //Sets end focus as last element
+        this.currentFocus = this.popoverMenuEls.length - 1;
+        this.setButtonFocus();
+        break;
+      case "Escape":
+      case "Tab":
+        this.closeMenu();
+        this.host.blur();
+        break;
+    }
+  }
+
+  private addMenuItems = (elements: string | any[] | NodeListOf<ChildNode>) => {
+    for (let i = 0; i < elements.length; i++) {
+      const el = elements[i] as HTMLIcMenuItemElement;
+      if (el.tagName === "IC-MENU-ITEM") {
+        this.popoverMenuEls.push(el);
+      } else if (el.tagName === "IC-MENU-GROUP") {
+        const groupSlotWrapper = el.shadowRoot.querySelector("ul");
+        const menuGroupElements = getSlotElements(groupSlotWrapper);
+
+        this.addMenuItems(menuGroupElements);
+      }
+    }
+  };
+
+  private getMenuAriaLabel = (): string => {
+    const ariaLabel = this.host.getAttribute("aria-label");
+
+    if (this.submenuId !== undefined) {
+      return `${ariaLabel}, within nested level ${this.submenuLevel} ${this.parentLabel} submenu,`;
+    } else {
+      return ariaLabel;
+    }
+  };
+
+  componentDidRender(): void {
+    if (this.open) {
+      createPopper(this.anchorEl, this.host, {
+        placement: "bottom-start",
+        modifiers: [
+          {
+            name: "offset",
+            options: {
+              offset: [0, 4],
+            },
+          },
+        ],
+      });
+    }
+  }
+
+  componentWillRender(): void {
+    this.anchorEl = this.findAnchorEl(this.anchor);
+  }
+
+  componentDidLoad(): void {
+    const slotWrapper = this.host.shadowRoot.querySelector("ul.button");
+    const popoverMenuElements = getSlotElements(slotWrapper);
+
+    if (popoverMenuElements !== null) {
+      this.addMenuItems(popoverMenuElements);
+    }
+
+    if (
+      this.submenuId === undefined &&
+      this.host.getAttribute("aria-label") === null
+    ) {
+      console.error(
+        `No aria-label specified for popover menu component - aria-label required`
+      );
+    }
+  }
+
+  render() {
+    const onBackButtonClick = (): void => {
+      this.parentPopover.openFromChild();
+      this.open = false;
+    };
+    return (
+      <Host>
+        <div
+          id={
+            this.parentPopover === undefined
+              ? `ic-popover-submenu-${this.submenuId}`
+              : ""
+          }
+          class={{
+            menu: true,
+          }}
+          tabindex={open ? "0" : "-1"}
+        >
+          <div
+            class={{
+              "opening-from-parent": this.openingFromParent,
+              "opening-from-child": this.openingFromChild,
+            }}
+          >
+            {this.submenuId !== undefined && (
+              <div>
+                <ic-menu-item
+                  class="ic-popover-submenu-back-button"
+                  ref={(el) => (this.backButton = el)}
+                  label="Back"
+                  onClick={onBackButtonClick}
```

Comment:
Let me know if it's still erroring, I can't see an error or warning for this, but it might be fixed just from moving it outside of the render and calling it like this.handleBackButtonClick